src/components/FormStepper.tsx

```diff
@@ -1,44 +1,51 @@
+import { useLocation } from 'react-router-dom';
 
 function FormStepper() {
-  
-    return (
+  const location = useLocation(); 
 
-        <div>
-        <div className="steps-sidebar">
-          <div className="step active">
-            <div className="step-number">1</div>
-            <div className="step-label">
-              <span className="step-title">STEP 1</span>
-              <span className="step-name">YOUR INFO</span>
-            </div>
-          </div>
-  
-          <div className="step">
-            <div className="step-number">2</div>
-            <div className="step-label">
-              <span className="step-title">STEP 2</span>
-              <span className="step-name">SELECT PLAN</span>
-            </div>
-          </div>
-  
-          <div className="step">
-            <div className="step-number">3</div>
-            <div className="step-label">
-              <span className="step-title">STEP 3</span>
-              <span className="step-name">ADD-ONS</span>
-            </div>
-          </div>
-  
-          <div className="step">
-            <div className="step-number">4</div>
-            <div className="step-label">
-              <span className="step-title">STEP 4</span>
-              <span className="step-name">SUMMARY</span>
-            </div>
-          </div>
-        </div>
-        </div>
-    );
-  }
-  
-export default FormStepper;
\ No newline at end of file
+  // Step component as props
+  const Step = ({ isActive, title, subText, stepNumber }) => (
+    <div className={`step ${isActive ? 'active' : ''}`}>
+      <div className="step-number">{stepNumber}</div>
+      <div className="step-label">
+        <span className="step-title">{title}</span>
+        <span className="step-name">{subText}</span>
+      </div>
+    </div>
+  );
+
+  return (
+    <div className="steps-sidebar">
+      {/* Step 1 */}
+      <Step
+        isActive={location.pathname === '/page1'} 
+        title="STEP 1"
+        subText="YOUR INFO"
+        stepNumber="1"
+      />
+      {/* Step 2 */}
+      <Step
+        isActive={location.pathname === '/page2'}
+        title="STEP 2"
+        subText="SELECT PLAN"
+        stepNumber="2"
+      />
+      {/* Step 3 */}
+      <Step
+        isActive={location.pathname === '/page3'} 
+        title="STEP 3"
+        subText="ADD-ONS"
+        stepNumber="3"
+      />
+      {/* Step 4 */}
+      <Step
+        isActive={location.pathname === '/page4'}
+        title="STEP 4"
+        subText="SUMMARY"
+        stepNumber="4"
+      />
+    </div>
```

Comment:
creating an array of objects will be way better, give it the name and URL, then comparing if it's active will be done in a single place